app/assets/javascripts/app/collections/notifications.js

```diff
@@ -0,0 +1,109 @@
+app.collections.Notifications = Backbone.Collection.extend({
+  model: app.models.Notification,
+  // URL parameter
+  /* eslint-disable camelcase */
+  url: Routes.notifications({per_page: 10, page: 1}),
+  /* eslint-enable camelcase */
+  page: 2,
+  perPage: 5,
+  unreadCount: 0,
+  unreadCountByType: {},
+
+  initialize: function() {
+    this.pollNotifications();
+    Diaspora.BrowserNotification.requestPermission();
+  },
+
+  pollNotifications: function() {
+    var unreadCountBefore = this.unreadCount;
+    this.fetch();
+    if (unreadCountBefore < this.unreadCount) {
+      Diaspora.BrowserNotification.spawnNotification(Diaspora.I18n.t("notifications.new_notifications"));
+    }
```

Comment:
At least while testing this feature it didn't work because the check `unreadCountBefore < this.unreadCount` was done before `fetch` finished. You can fix this by listening to the `finishedLoading` event:

```diff
   pollNotifications: function() {
     var unreadCountBefore = this.unreadCount;
+    this.once("finishedLoading", function() {
+      if (unreadCountBefore < this.unreadCount) {
+        Diaspora.BrowserNotification.spawnNotification(Diaspora.I18n.t("notifications.new_notifications"));
+      }
+    }, this);
     this.fetch();
-    if (unreadCountBefore < this.unreadCount) {
-      Diaspora.BrowserNotification.spawnNotification(Diaspora.I18n.t("notifications.new_notifications"));
-    }
```

This would also show a browser notification immediately after loading the page when there are unread notifications. To fix this call `this.fetch()` in `initialize` and call `pollNotifications` the first time after some timeout.